File: HotelDemo/src/app/models/room-dto.ts
export class RoomDTO {
  roomId: number;
  roomNumber: string;
  roomType: string;
  status: string;

  constructor(
    roomId: number = 0,
    roomNumber: string = '',
    roomType: string = '',
    status: string = ''
  ) {
    this.roomId = roomId;
    this.roomNumber = roomNumber;
    this.roomType = roomType;
    this.status = status;
  }
}

File: DemoTwo/DemoTwo/DTOs/RoomDTO.cs
using System.ComponentModel.DataAnnotations;

namespace DemoTwo.DTOs
{
    public class RoomDTO
    {
        public int RoomId { get; set; }

        [Required]
        [StringLength(10)]
        public string RoomNumber { get; set; }

        [Required]
        [StringLength(50)]
        public string RoomType { get; set; }

        [Required]
        [StringLength(20)]
        public string Status { get; set; }
    }
}
